server.py

```diff
@@ -0,0 +1,343 @@
+                image_form_data.get('Image Tag'), image_form_data.get('Category'), session['user'])
+
+    path = UPLOAD_FOLDER + "/" + filename
+
+    converted_image = Image.open(path)
+    converted_image.save(os.path.join(app.config['UPLOAD_FOLDER'], temp_file_name+'.png'))
+    print('data posted')
+    image_list = get_image_data()
+    print('data sending')
+    return jsonify(image_list)
+
+
+def get_searched_data(key):
+    """
+    This function is searching Images on the basis of key.
+    :param key: Key to match the data.
+    :return: Returns list of images whose tag, title and description contains key.
+    """
+    key = str(key)
+    result = Images.query.filter(or_(Images.title.contains(key), Images.tag.contains(key),
+                                     Images.description.contains(key))).all()
+    image_list = []
+    for row in result:
+        image_list.append(row.serialize(session['user']))
+    return image_list
+
+
+def get_category_data(key):
+    """
+    Searching Images on the basis of category.
+    :param key: Category
+    :return: Returns list of images whose category matches key.
+    """
+    key = str(key.strip())
+    if key != "All":
+        result = Images.query.filter_by(category=key).all()
+    else:
+        result = Images.query.order_by(desc(Images.id)).all()
+    image_list = []
+    for row in result:
+        image_list.append(row.serialize(session['user']))
+    return image_list
+
+
+@app.route('/search', methods=['GET'])
+def search():
+    """
+    Search route
+    """
+    key = request.args.get('key')
+    result = get_searched_data(key)
+    return jsonify(result)
+
+
+@app.route('/category', methods=['GET'])
+def category_search():
+    """
+    Search category route
+    """
+    key = request.args.get('category')
+    result = get_category_data(key)
+    print result
+    return jsonify(result)
+
+
+def email_existence(email):
+    """
+    This function is checking either an email exist, during signup.
+    :param email: email to be checked
+    """
+    result = Users.query.filter_by(email=email).first()
+    if result is None:
+        print "No email exist"
+        return True
+    else:
+        print "email exist"
+        return False
+
+
+@app.route('/email_availability/', methods=['GET'])
+def email_availability():
+    """
+    Email availability route
+    """
+    email = request.args.get('email')
+    print(email)
+    if email_existence(str(email)) is True:
+        return "true"
+    else:
+        return "false"
+
+
+def register_user(name, password, email):
+    """
+    This function is registering the user.
+    :param name: Name of user
+    :param password: Password of user
+    :param email: Email of user
+    """
+    name = str(name)
+    password = str(password)
+    email = str(email)
+    new_user = Users(name=name, password=password, email=email)
+    db.session.add(new_user)
+    db.session.commit()
+
+
+@app.route('/login', methods=['GET'])
```

Comment:
I think 2 routes are rendering same response, right? `/` and `/login`. If so, we can combine both views in one.